src/gui/src/staDescriptors.cpp

```diff
@@ -0,0 +1,498 @@
+    }
+  }
+};
+
+//////////////////////////////////////////////////
+
+LibertyLibraryDescriptor::LibertyLibraryDescriptor(odb::dbDatabase* db,
+                                                   sta::dbSta* sta)
+    : db_(db), sta_(sta)
+{
+}
+
+std::string LibertyLibraryDescriptor::getName(std::any object) const
+{
+  return std::any_cast<sta::LibertyLibrary*>(object)->name();
+}
+
+std::string LibertyLibraryDescriptor::getTypeName() const
+{
+  return "Liberty library";
+}
+
+bool LibertyLibraryDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  return false;
+}
+
+void LibertyLibraryDescriptor::highlight(std::any object,
+                                         Painter& painter) const
+{
+}
+
+Descriptor::Properties LibertyLibraryDescriptor::getProperties(
+    std::any object) const
+{
+  auto library = std::any_cast<sta::LibertyLibrary*>(object);
+
+  Properties props;
+
+  auto gui = Gui::get();
+  SelectionSet cells;
+  sta::LibertyCellIterator cell_iter(library);
+  while (cell_iter.hasNext()) {
+    cells.insert(gui->makeSelected(cell_iter.next()));
+  }
+  props.push_back({"Cells", cells});
+
+  return props;
+}
+
+Selected LibertyLibraryDescriptor::makeSelected(std::any object) const
+{
+  if (auto library = std::any_cast<sta::LibertyLibrary*>(&object)) {
+    return Selected(*library, this);
+  }
+  return Selected();
+}
+
+bool LibertyLibraryDescriptor::lessThan(std::any l, std::any r) const
+{
+  auto l_library = std::any_cast<sta::LibertyLibrary*>(l);
+  auto r_library = std::any_cast<sta::LibertyLibrary*>(r);
+  return l_library->id() < r_library->id();
+}
+
+bool LibertyLibraryDescriptor::getAllObjects(SelectionSet& objects) const
+{
+  sta::dbNetwork* network = sta_->getDbNetwork();
+  std::unique_ptr<sta::LibertyLibraryIterator> lib_iter{
+      network->libertyLibraryIterator()};
+
+  while (lib_iter->hasNext()) {
+    sta::LibertyLibrary* library = lib_iter->next();
+    objects.insert(makeSelected(library));
+  }
+
+  return true;
+}
+
+//////////////////////////////////////////////////
+
+LibertyCellDescriptor::LibertyCellDescriptor(odb::dbDatabase* db,
+                                             sta::dbSta* sta)
+    : db_(db), sta_(sta)
+{
+}
+
+std::string LibertyCellDescriptor::getName(std::any object) const
+{
+  return std::any_cast<sta::LibertyCell*>(object)->name();
+}
+
+std::string LibertyCellDescriptor::getTypeName() const
+{
+  return "Liberty cell";
+}
+
+bool LibertyCellDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  return false;
+}
+
+void LibertyCellDescriptor::highlight(std::any object, Painter& painter) const
+{
+}
+
+Descriptor::Properties LibertyCellDescriptor::getProperties(
+    std::any object) const
+{
+  auto cell = std::any_cast<sta::LibertyCell*>(object);
+
+  auto gui = Gui::get();
+  Properties props;
+
+  if (auto area = cell->area()) {
+    props.push_back({"Area", area});
+  }
+  add_if_true(props, "Dont Use", cell->dontUse());
```

Comment:
I'm not sure if the `add_if_true` makes sense to me. In all other descriptors we've added the false value as well